src/transformers/models/speecht5/modeling_speecht5.py

```diff
@@ -608,23 +608,20 @@ def forward(
 
         return hidden_states, attention_mask
 
-    # Copied from transformers.models.unispeech.modeling_unispeech.UniSpeechPreTrainedModel._get_feature_vector_attention_mask
+    # Copied from transformers.models.hubert.modeling_hubert.HubertPreTrainedModel._get_feature_vector_attention_mask
     def _get_feature_vector_attention_mask(self, feature_vector_length: int, attention_mask: torch.LongTensor):
-        # Effectively attention_mask.sum(-1), but not inplace to be able to run
```

Comment:
@sanchit-gandhi Do you know if .sum(-1) still cause problems? It's introduced in #14260. I try change `.cumsum(-1)[:, -1]` to `.sum(-1)` in wav2vec2 and all tests still pass. I think it will be best to consolidate similar functions to wav2vec2. Before my PR, Hubert's version uses .sum(-1)

Reply:
Yes I believe it does - ideally let's keep it as the `.cumsum` op to avoid the in-place operations

Reply:
Currently I see two variants of this this function
1. Wav2Vec2, Wav2Vec2Conformer, WavLM, Data2Vec-Audio: there is `add_adapter` argument
2. HuBERT, SEW, SEW-D, UniSpeech, UniSpeech-SAT, and SpeechT5: no `add_adapter` argument

Do you know the reason why there are two variants? `add_adapter` is always False anyway from what I see. Ideally I think we should consolidate everything to Wav2Vec2 version (make other models copy from Wav2Vec2), since all other models are modifications from Wav2Vec2.

Reply:
`add_adapter` is used to account for the linear adapter layer that is optionally added on top of an audio encoder model. This adapter layer further downsamples along the temporal dimension of the `hidden_states`, _c.f._ https://github.com/huggingface/transformers/tree/main/examples/pytorch/speech-recognition#warm-started-speech-encoder-decoder-model

This adapter is only used in a handful of models, e.g. when the model is a candidate encoder in an encoder-decoder configuration

Reply:
I see. Would it be okay if I use Wav2Vec2's `_get_feature_vector_attention_mask()`  for all other models, even if they don't have adapters? Having one source of truth would be easier for maintenance.

Reply:
I would have two sets: one for models with adapters (e.g. Wav2Vec2), one for models without (e.g. HuBERT). All the models with adapters should copy from Wav2Vec2. All the models without should copy from HuBERT. Otherwise, the models without adapters have redundant adapter code

Reply:
I had done what you suggested before this comment :)